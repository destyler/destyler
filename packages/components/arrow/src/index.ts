export { DestylerArrow, type DestylerArrowProps } from './arrow'
