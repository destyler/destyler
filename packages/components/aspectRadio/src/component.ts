export { DestylerAspectRadio, destylerAspectRadioProps } from './components/aspectRadio'
export type { DestylerAspectRadioProps } from './components/aspectRadio'
