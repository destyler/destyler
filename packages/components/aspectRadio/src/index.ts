export * from './aspectRadio'
export * from './resolver'
