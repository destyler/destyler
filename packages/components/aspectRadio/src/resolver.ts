import type { ComponentResolver } from 'unplugin-vue-components'
import { name } from '../package.json'

const packageName = name

export const aspectRadioComponentName = [
  'AspectRadio',
]

export function DestylerAspectRadioResolver(): ComponentResolver {
  return {
    type: 'component',
    resolve: (name: string) => {
      if (aspectRadioComponentName.includes(name))
        return { name, from: packageName }
    },
  }
}
