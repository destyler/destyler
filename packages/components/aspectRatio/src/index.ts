export * from './aspectRatio'
export * from './resolver'
