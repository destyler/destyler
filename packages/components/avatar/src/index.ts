export * from './avatarRoot'
export * from './avatarImage'
export * from './avatarFallback'
export * from './resolver'
