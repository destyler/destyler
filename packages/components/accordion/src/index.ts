import { DestylerAccordionResolver } from './resolver'
import DestylerAccordionRoot from './AccordionRoot.vue'
import DestylerAccordionItem from './AccordionItem.vue'
import { selected, useAccordionSelected } from './composition'
import destylerAccordionPresets from './presets'

export * from './props'

export {
  selected,
  DestylerAccordionRoot,
  DestylerAccordionItem,
  DestylerAccordionResolver,
  useAccordionSelected,
  destylerAccordionPresets,
}
