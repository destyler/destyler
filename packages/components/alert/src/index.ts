import { DestylerAccordionResolver } from './resolver'
import DestylerAlert from './Alert'

export {
  DestylerAlert,
  DestylerAccordionResolver,
}
