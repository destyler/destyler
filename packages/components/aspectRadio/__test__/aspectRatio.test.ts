import { beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'
import type { VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import { DestylerAspectRadio } from '../src'

describe('aspectRatio', async () => {
  let wrapper: VueWrapper<InstanceType<typeof DestylerAspectRadio>>

  beforeEach(() => {
    wrapper = mount(DestylerAspectRadio, {
      slots: {
        default: `<img src="https://cdn.vuetifyjs.com/images/parallax/material.jpg" class="h-full w-full object-cover" alt="Destyler Image">`,
      },
    })
  })

  it('axe accessibility', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  it('render as snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
