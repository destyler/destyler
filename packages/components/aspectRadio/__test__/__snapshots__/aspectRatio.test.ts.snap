// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aspectRatio > render as snapshot 1`] = `
"<div style="position: relative; width: 100%; padding-bottom: 100%;" data-destyler-aspect-ratio-wrapper="">
  <div style="position: absolute; inset: 0px;"><img src="https://cdn.vuetifyjs.com/images/parallax/material.jpg" class="h-full w-full object-cover" alt="Destyler Image"></div>
</div>"
`;

exports[`given a default AspectRatio > should render as snapshot 1`] = `
"<div style="position: relative; width: 100%; padding-bottom: 100%;" data-destyler-aspect-ratio-wrapper="">
  <div style="position: absolute; inset: 0px;"><img src="https://cdn.vuetifyjs.com/images/parallax/material.jpg" class="h-full w-full object-cover" alt="Destyler Image"></div>
</div>"
`;
